yt_dlp/YoutubeDL.py

```diff
@@ -702,8 +707,7 @@ def check_deprecated(param, option, suggestion):
                 get_postprocessor(pp_def.pop('key'))(self, **pp_def),
                 when=when)
 
-        self._setup_opener()
-        register_socks_protocols()
+        self.http = self.build_http(REQUEST_HANDLERS)
```

Comment:
confusing name

Reply:
yeah I'm not sure what to name this, open for suggestions

request_broker
session
http_session
network_session
client
http_client

Reply:
Probly `_request_broker`. It's not exactly a client or session

Reply:
`_request_handler` conflicts with RequestHandler

I've changed it to `_request_director` for now, as I've renamed the broker to RequestDirector